docs/user/non-blocking-prefetch.md

```diff
@@ -0,0 +1,406 @@
+
+- OS image downloads
+- Application container image downloads
+- OCI artifact downloads for application volumes
+
+### Monitoring Prefetch Status
+
+You can monitor prefetch progress through device status:
+
+```yaml
+status:
+  updated:
+    status: OutOfDate              # Update available but not yet applied
+  conditions:
+    - type: DeviceUpdating
+      status: "True"
+      reason: Preparing
+      message: "Downloading OS image in background"
+```
+
+Common prefetch status messages:
+
+- `"Downloading OS image in background"` - OS image download in progress
+- `"Prefetch not ready"` - Downloads still in progress or failed
+- `"Ready to apply update"` - All required images have been downloaded
+
+## Best Practices
+
+### Download Optimization
+
+1. **Appropriate Timeouts**: Configure pull timeouts based on your network conditions and image sizes:
+
+   ```yaml
+   # For slow networks or large images
+   pull-timeout: 30m
+   
+   # For fast networks or smaller images  
+   pull-timeout: 10m
+   ```
+
+2. **Network Considerations**:
+    - Schedule downloads during off-peak hours when possible
+    - Account for multiple devices downloading simultaneously
+    - Consider bandwidth limitations at edge locations
+
+### Disk Space Management
+
+1. **Monitor Available Space**: Ensure adequate disk space for image downloads:
+    - OS images can be 1-2 GB or larger
+    - Application images add additional space requirements
+    - Allow for temporary space during downloads
+
+2. **Regular Cleanup**: Clean up unused images periodically:
+
+   ```bash
+   podman system prune -a
+   ```
+
+3. **Storage Planning**: Plan storage capacity for:
+    - Current OS image
+    - New OS image during updates
+    - Application container images
+    - Temporary download space
+
+## Troubleshooting
+
+### Common Issues and Solutions
+
+#### Update Stuck in "Preparing" State
+
+**Symptom**: Device shows `Preparing` state for extended periods
+
+**Potential Causes**:
+
+- Large OS image downloads taking time to complete
+- Network connectivity issues during download
+- Insufficient disk space preventing download completion
+- Download timeouts causing repeated retries
+
+**Resolution**:
+
+1. Check agent logs for download progress:
+
+   ```bash
+   journalctl -u flightctl-agent | grep -i prefetch
+   ```
+
+2. Check available disk space:
+
+   ```bash
+   df -h
+   ```
+
+3. Verify network connectivity to registry:
+
+   ```bash
+   ping registry.example.com
+   ```
+
+4. Check if images are being downloaded:
+
+   ```bash
+   podman images
+   ```
+
+#### "Prefetch Not Ready" Errors
+
+**Symptom**: Updates fail with "oci prefetch not ready" errors
+
+**Potential Causes**:
+
+- Downloads still in progress
+- Download failures requiring retry
+- Network issues preventing downloads
+
+**Resolution**:
+
+1. Wait for current downloads to complete (check logs for progress)
+2. Clean up disk space if low:
+
+   ```bash
+   podman system prune -a
```

Comment:
one caveat here is prune will remove all the images we have downloaded but not yet used. so it would require redownloading.